ui/src/channels/NewChannel/NewChannelForm.tsx

```diff
@@ -39,7 +39,22 @@ export default function NewChannelForm() {
   const onSubmit = useCallback(
     async (values: NewChannelFormSchema) => {
       const { privacy, type, ...nextChannel } = values;
-      const channelName = strToSym(values.meta.title);
+      /*
+        For now channel names are used as keys for pacts. Therefore we need to
+        check if a channel with the same name already exists in the chat store. If it does, we
+        need to append a timestamp to the end of the name of the new channel.
+
+        Timestamps are used because they are virtually guaranteed to be unique.
+
+        In the future, we will index channels by their full path (including group name), and this will no
+        longer be necessary. That change will require a migration of existing channels.
+       */
+      const tempChannelName = strToSym(values.meta.title);
+      const tempNewChannelFlag = `${window.our}/${tempChannelName}`;
+      const existingChannel = useChatState.getState().pacts[tempNewChannelFlag];
```

Comment:
We actually need to check the channel type that matches the type of channel trying to be created, not just chat. Also a more reliable place I think would be to check `chats` not `pacts` even though they should both have the same keys.

Reply:
Good point, just pushed a commit addressing this (and using chats instead of pacts).